src/box/memtx_engine.c

```diff
@@ -205,9 +211,9 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	 */
 	if (!xlog_cursor_is_eof(&cursor)) {
 		if (!memtx->force_recovery)
-			panic("snapshot `%s' has no EOF marker", filename);
```

Comment:
If we reach so far, why not use cursor->name instead of filename? xlog_cursor_close() even has comment saying that it is ok to address to cursor's fields even after closing it.